server/src/main/java/com/hedera/block/server/producer/ProducerBlockItemObserver.java

```diff
@@ -108,28 +115,59 @@ public void onNext(@NonNull final PublishStreamRequest publishStreamRequest) {
             @NonNull final var errorResponse = buildErrorStreamResponse();
             publishStreamResponseObserver.onNext(errorResponse);
             LOGGER.log(System.Logger.Level.ERROR, "Exception thrown publishing BlockItem: ", io);
-
             LOGGER.log(System.Logger.Level.ERROR, "Shutting down the web server");
             serviceStatus.stopWebServer();
+        } catch (ParseException e) {
+            @NonNull final var errorResponse = buildErrorStreamResponse();
+            publishStreamResponseObserver.onNext(errorResponse);
+            LOGGER.log(
+                    System.Logger.Level.ERROR,
+                    "Error parsing inbound block item from a producer: "
+                            + publishStreamRequest.getBlockItem(),
+                    e);
+            serviceStatus.stopWebServer();
         }
     }
 
     @NonNull
-    private PublishStreamResponse buildSuccessStreamResponse(@NonNull final BlockItem blockItem)
-            throws IOException, NoSuchAlgorithmException {
-        @NonNull final ItemAcknowledgement itemAck = itemAckBuilder.buildAck(blockItem);
-        return PublishStreamResponse.newBuilder().setAcknowledgement(itemAck).build();
+    private com.hedera.hapi.block.protoc.PublishStreamResponse buildSuccessStreamResponse(
```

Comment:
Perhaps worth noting that this will be much more complex when complete, and likely we'll need to merge success or failure into a single method that takes a status, as well as other data (and allows null for most parameters).
There will also be more different response types (e.g. block or item ack, or one of several end of stream codes, including success, timeout, "behind" etc...)

Reply:
Makes sense. There's definitely a lot of protocol integration work to come.